Args.cpp

```diff
@@ -106,7 +106,7 @@ void Args::parse(string line) {
     else if (key == "-user") { user = s; }
     else if (key == "-cpu") { cpu = s; }
     else if (key == "-time") { timeKernels = true; }
-    else if (key == "-device" || key == "-d") { device = stoi(s); }
+    else if (key == "-device" || key == "-d") { device = stoi(s); cpu = getHwName(device) + "-" + getBoardName(device) + "-" + str(device); }
```

Comment:
Here we may be overwriting a user specified -cpu, and we're only setting it when -d is specified.
A first step could be to only set cpu here if it's empty, i.e. if (cpu.empty()) { cpu = ... },
and do it on separate lines please.